modules/internal/ChapelArray.chpl

```diff
@@ -4016,14 +4016,23 @@ module ChapelArray {
   }
 
   pragma "init copy fn"
-  proc chpl__initCopy(const ref a: []) {
+  proc chpl__initCopy(a: []) {
     var b : [a._dom] a.eltType;
 
+    if !isConstAssignableType(a.eltType) {
+      if !isAssignableType(a.eltType) {
+        compilerError("Cannot copy array with element type that cannot be assigned");
+      } else {
+        // Make sure this function uses ref argument intent for a
+        ref r = a;
```

Comment:
This seems pretty hacky, but I'm OK with it for now. Could it be rewritten to use where-clauses?

Reply:
Yes, it could. I wrote it this way on purpose to avoid a where clause in order to (potentially) reduce compile time. I didn't measure the impact though.

I'll leave this for now but update the comment to say it would be cleaner to have a where clause.